### PR TITLE
feat (provider/google-vertex): Add support for Anthropic models.

### PR DESCRIPTION
**Pull request by shaper**
- [x] decide on sub-module names under `@ai-sdk/google-vertex`: `.../anthropic-edge` and `.../anthropic-node` vs `.../anthropic` and `.../anthropic/edge`
- [x] add/review provider package tests
- [x] add/review e2e tests
- [x] rework provider settings as needed to reflect validation results for cache control, pdf support, etc.
- [x] consider default provider name -- `googleVertexAnthropic` or `vertexAnthropic`
- [x] update docs

### CODE REVIEW

**Comment by lgrammel** [comment]
Looks great!

File: packages/anthropic/src/anthropic-messages-language-model.ts [hunk 1]
```diff
@@ -179,21 +183,32 @@ export class AnthropicMessagesLanguageModel implements LanguageModelV1 {
...
   }
 
+  private buildRequestUrl(isStreaming: boolean): string {
+    return (
+      this.config.buildRequestUrl?.(this.config.baseURL, isStreaming) ??
+      `${this.config.baseURL}/messages`
+    );
+  }
+
+  private transformArgs(args: Record<string, any>): Record<string, any> {
```
Comment (lgrammel) [inline packages/anthropic/src/anthropic-messages-language-model.ts:199]: `transformArgs` had me wondering at first what it does. Based on the usage below, I'm wondering if `transformBody` or `transformRequestBody` would be a more intuitive name for both this method and the config property?

File: packages/google-vertex/src/anthropic/edge/google-vertex-anthropic-provider-edge.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export function createVertexAnthropic(
+  options: GoogleVertexAnthropicProviderSettings = {},
+): GoogleVertexAnthropicProvider {
+  return createVertexAnthropicOriginal({
+    ...options,
+    headers:
+      options.headers ??
+      (async () => ({
+        'anthropic-version': 'vertex-2023-10-16',
```
Comment (lgrammel) [inline packages/google-vertex/src/anthropic/edge/google-vertex-anthropic-provider-edge.ts:31]: is the anthropic version needed in both headers and body?

**Review by lgrammel — APPROVED** [review]

File: packages/google-vertex/src/anthropic/edge/google-vertex-anthropic-provider-edge.ts (hunk 2 above)
Comment (shaper) [reply packages/google-vertex/src/anthropic/edge/google-vertex-anthropic-provider-edge.ts:31]: Good find, it appears we need `anthropic_version: 'vertex-2023-10-16',` in the request body but we don't need this in the headers. Removed here and in node variant.

File: packages/anthropic/src/anthropic-messages-language-model.ts (hunk 1 above)
Comment (shaper) [reply packages/anthropic/src/anthropic-messages-language-model.ts:199]: Agreed, renamed to `transformRequestBody`.

**Pull request merged**